internal/usecase/profiles/usecase.go

```diff
@@ -108,9 +118,38 @@ func (uc *UseCase) Delete(ctx context.Context, profileName, tenantID string) err
 	return nil
 }
 
+func (uc *UseCase) WifiProfileExistsfunc(ctx context.Context, d *dto.Profile, action string) error {
+	if len(d.WiFiConfigs) > 0 {
+		// check if the wireless profile is exists in the database
+		wifiProfiles := []string{}
+
+		for _, wifiConfig := range d.WiFiConfigs {
+			result, err := uc.wifiConfig.CheckProfileExists(ctx, wifiConfig.WirelessProfileName, d.TenantID)
+			if err != nil {
+				return err
+			}
+
+			if !result {
+				wifiProfiles = append(wifiProfiles, wifiConfig.WirelessProfileName)
+			}
+		}
+
+		if len(wifiProfiles) > 0 {
+			return ErrNotValid.Wrap(action, "uc.wifiConfig.CheckProfileExists", consoleerrors.CreateConsoleError("wifiProfiles are not found in the database"))
+		}
+	}
+
+	return nil
+}
+
 func (uc *UseCase) Update(ctx context.Context, d *dto.Profile) (*dto.Profile, error) {
 	d1 := uc.dtoToEntity(d)
 
+	err := uc.WifiProfileExistsfunc(ctx, d, "update")
```

Comment:
why does this have func in the name? of course its a func....

Reply:
done